File: src/class/Varietal.js
export default class Varietal {
  constructor(descripcion, porcentajeComposicion, tipoDeUva) {
    this.descripcion = descripcion;
    this.porcentajeComposicion = porcentajeComposicion;
    this.tipoDeUva = tipoDeUva; // Relación con la clase TipoDeUva
  }

  // Métodos getter y setter para los atributos de la clase Varietal

  get descripcion() {
    return this.descripcion;
  }

  set descripcion(descripcion) {
    this.descripcion = descripcion;
  }

  get porcentajeComposicion() {
    return this.porcentajeComposicion;
  }

  set porcentajeComposicion(porcentajeComposicion) {
    this.porcentajeComposicion = porcentajeComposicion;
  }

  get tipoDeUva() {
    return this.tipoDeUva;
  }

  set tipoDeUva(tipoDeUva) {
    this.tipoDeUva = tipoDeUva;
  }
}

File: src/com/BonVino/proyect/Varietal.java
package com.BonVino.proyect;

public class Varietal {
    private String descripcion;
    private double porcentajeComposicion;
    private TipoDeUva tipoDeUva; // Relación con la clase TipoDeUva

    // Constructor
    public Varietal(String descripcion, double porcentajeComposicion, TipoDeUva tipoDeUva) {
        this.descripcion = descripcion;
        this.porcentajeComposicion = porcentajeComposicion;
        this.tipoDeUva = tipoDeUva;
    }

    // Métodos getter y setter para descripcion
    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    // Métodos getter y setter para porcentajeComposicion
    public double getPorcentajeComposicion() {
        return porcentajeComposicion;
    }

    public void setPorcentajeComposicion(double porcentajeComposicion) {
        this.porcentajeComposicion = porcentajeComposicion;
    }

    // Métodos getter y setter para tipoDeUva
    public TipoDeUva getTipoDeUva() {
        return tipoDeUva;
    }

    public void setTipoDeUva(TipoDeUva tipoDeUva) {
        this.tipoDeUva = tipoDeUva;
    }
}
